android/libraries/rib-badoo/src/main/java/com/badoo/common/rib/routing/backstack/BackStackManager.kt

```diff
@@ -0,0 +1,186 @@
+package com.badoo.common.rib.routing.backstack
+
+import android.os.Parcelable
+import com.badoo.common.rib.routing.RibConnector
+import com.badoo.common.rib.routing.action.RoutingAction
+import com.badoo.common.rib.routing.backstack.BackStackManager.Action
+import com.badoo.common.rib.routing.backstack.BackStackManager.Action.Execute
+import com.badoo.common.rib.routing.backstack.BackStackManager.Action.ReattachRibsOfLastEntry
+import com.badoo.common.rib.routing.backstack.BackStackManager.Effect
+import com.badoo.common.rib.routing.backstack.BackStackManager.State
+import com.badoo.common.rib.routing.backstack.BackStackManager.Wish
+import com.badoo.common.rib.routing.backstack.BackStackManager.Wish.NewRoot
+import com.badoo.common.rib.routing.backstack.BackStackManager.Wish.Pop
+import com.badoo.common.rib.routing.backstack.BackStackManager.Wish.Push
+import com.badoo.common.rib.routing.backstack.BackStackManager.Wish.Replace
+import com.badoo.common.rib.routing.backstack.BackStackManager.Wish.ShrinkToBundles
+import com.badoo.common.rib.routing.backstack.BackStackManager.Wish.TearDown
+import com.badoo.common.rib.routing.backstack.BackStackRibConnector.DetachStrategy.DESTROY
+import com.badoo.common.rib.routing.backstack.BackStackRibConnector.DetachStrategy.DETACH_VIEW
+import com.badoo.mvicore.element.Actor
+import com.badoo.mvicore.element.Bootstrapper
+import com.badoo.mvicore.element.Reducer
+import com.badoo.mvicore.element.TimeCapsule
+import com.badoo.mvicore.feature.BaseFeature
+import io.reactivex.Observable
+import io.reactivex.Observable.empty
+import io.reactivex.Observable.just
+import kotlinx.android.parcel.Parcelize
+
+internal class BackStackManager<C : Parcelable>(
+    resolver: (C) -> RoutingAction<*>,
+    ribConnector: RibConnector,
+    initialConfiguration: C,
+    timeCapsule: TimeCapsule<State<C>>,
+    tag: String
+): BaseFeature<Wish<C>, Action<C>, Effect<C>, State<C>, Nothing>(
```

Comment:
If you use delegation like ": Feature<Wish, State, Nothing> by BaseFeature<...>(...)" then you can make Actions, Effects, Actor, Reducer and Bootstrapper private, which will provide better encapsulation.

Reply:
The whole class is internal, not an issue I guess.